lib/awful/tag.lua

```diff
@@ -431,7 +434,19 @@ end
 -- @param add Value to add to number of master windows.
 -- @param t The tag to modify, if null tag.selected() is used.
 function tag.incnmaster(add, t)
-    tag.setnmaster(tag.getnmaster(t) + add, t)
+    client = client or require("awful.client")
+    local ntiled = #client.tiled(1)
```

Comment:
That should be the tag's screen probably (assuming that `1` is the screen here).